accounts-db/src/tiered_storage/hot.rs

```diff
@@ -457,25 +483,130 @@ impl HotStorageWriter {
             storage: TieredStorageFile::new_writable(file_path)?,
         })
     }
+
+    /// Persists an account with the specified information and returns
+    /// the stored size of the account.
+    fn write_account(
+        &self,
+        lamports: u64,
+        rent_epoch: Epoch,
+        account_data: &[u8],
+        executable: bool,
+        hash: &AccountHash,
+    ) -> TieredStorageResult<usize> {
+        let optional_fields = AccountMetaOptionalFields {
+            rent_epoch: (rent_epoch != u64::MAX).then_some(rent_epoch),
+            account_hash: (*hash != AccountHash(Hash::default())).then_some(*hash),
+        };
+
+        let mut flags = AccountMetaFlags::new_from(&optional_fields);
+        flags.set_executable(executable);
+
+        let padding_len = padding_bytes(account_data.len());
+        let meta = HotAccountMeta::new()
+            .with_lamports(lamports)
+            .with_account_data_size(account_data.len() as u64)
+            .with_account_data_padding(padding_len)
+            .with_flags(&flags);
+
+        let mut stored_size = 0;
+
+        stored_size += self.storage.write_pod(&meta)?;
+        stored_size += self.storage.write_bytes(account_data)?;
+        stored_size += self
+            .storage
+            .write_bytes(&PADDING_BUFFER[0..(padding_len as usize)])?;
+        stored_size += write_optional_fields(&self.storage, &optional_fields)?;
+
+        Ok(stored_size)
+    }
+
+    /// A work-in-progress function that will eventually implements
+    /// AccountsFile::appends_account()
+    pub fn write_accounts<
+        'a,
+        'b,
+        T: ReadableAccount + Sync,
+        U: StorableAccounts<'a, T>,
+        V: Borrow<AccountHash>,
+    >(
+        &self,
+        accounts: &StorableAccountsWithHashesAndWriteVersions<'a, 'b, T, U, V>,
+        skip: usize,
```

Comment:
This is following the convention of the append_accounts API.   I think accounts db has a logic that in case not all the accounts have been written, then it will invoke another append_accounts in a different file with skip.

Reply:
The relevant code snipe can be found [here](https://github.com/solana-labs/solana/blob/master/accounts-db/src/accounts_db.rs#L6247-L6249).  Not sure whether we will eventually deprecate this skip, but for now using the same API as `append_accounts` will make the bridge easier.


```
// inside fn write_accounts_to_storage

         while infos.len() < accounts_and_meta_to_store.len() {
             let mut append_accounts = Measure::start("append_accounts");
             let rvs = storage
                 .accounts
                 .append_accounts(accounts_and_meta_to_store, infos.len());
```